api/views.py

```diff
@@ -152,14 +153,18 @@ class TaskListAPIView(ListAPIView):
     serializer_class = TaskSerializer
 
     def get_queryset(self):
-        selected = cache.get('published')
-        if selected is None:
-            queryset = Task.objects.all()
-            published_pks = [task.pk for task in queryset if task.is_published()]
-            queryset = Task.objects.filter(pk__in=published_pks)
-            cache.set('published', queryset, timeout=None)
-            return queryset
-        return selected
+        game = Game.objects.all()[0]
+
+        if 'game.add_task' in self.request.user.get_all_permissions() or \
```

Comment:
This statement lacks the 'else' branch. If the condition doesn't hold we should return an empty QuerySet.